server.go

```diff
@@ -135,7 +134,10 @@ func (s *server) Allow(ctx context.Context, namespace, name string, tokensReques
 		maxWaitTime *= time.Duration(b.Config().WaitTimeoutMillis)
 	}
 
-	w, success := b.Take(ctx, tokensRequested, maxWaitTime)
+	w, success, err := b.Take(ctx, tokensRequested, maxWaitTime)
+	if err != nil {
+		return 0, b.Dynamic(), errors.Wrap(err, "failed to take tokens")
```

Comment:
There's a comment below `// The only positive result` that is no longer accurate.

Reply:
It's still somewhat accurate, I tweaked it